### ISSUE TITLE
[Feature request] Ability to set matching tags for each reading status and automatically apply these as read status changes (for mobile tracking)

### ISSUE THREAD

**Issue by FeralFlora** [Dominic-DallOsto/zotero-reading-list#8]
As Zotero for iOS doesn't have addons, the reading status is not easily accessible on mobile devices. Perhaps a workaround would be if this plugin had the ability to apply tags that match the reading status. I know this is not a clean solution, but I think it's better than nothing.

**Comment by Dominic-DallOsto** [comment]
Just to go through in a bit more detail, this would need:
- On desktop - when you change the read status of an item a corresponding tag is automatically applied
- On iOS - you manually set the tags on items you've read
- When you return to desktop, items that have a different tag to their read status are updated (maybe after displaying a confirmation to the user)
  - Unfortunately I think we lose date information here? We don't know when the tag was set - if it was before or after the read status set on desktop?
- An option to apply the tag to all items with an existing read status

**Comment by MSBack** [comment]
The first two points correspond to what I was thinking.

To be honest, I didn't consider the situation going from iOS to desktop. I don't know enough about the Zotero backend and the data available to suggest a solution. 

The option to bulk add tags based on read status makes migrating to such a workflow much easier.

Perhaps a solution could be to make standard or custom rules on which statuses supersede others? For example. If the current status is "In progress" and the tag is "Read", it is clear that the status should be updated. The same goes for:
- "To read" -> "In progress"
- "To read" -> "Read"

"Not reading" is the most problematic one to reconcile, because it can change in any direction.

Do you know if the Zotero app will support extensions eventually?

**Comment by MSBack** [comment]
Hey, do you have any new thoughts on this?

**Comment by bruce-willis** [comment]
Just a ping to this issue.

I like to manage my collection via the desktop app, but I like to read papers on an iPad. Unfortunately, I could not filter by the "To read" or "In progress" property on a tablet. 

For a workaround, one can use the following extension: [zotero-actions-tags](https://github.com/windingwind/zotero-actions-tags) for automatic management of tags and use only tags for reading status, but I really like this extension and hope it will be possible to filter by status on mobile devices (even only one-way sync desktop -> tablet is better than nothing).

**Comment by Dominic-DallOsto** [comment]
I've started work on this - any thoughts would be great!

My idea at the moment is:
* to automatically set a tag when setting an item's read status
* update all items' tags to match read status
    * manual option
    * automatically when enabling the preference for the first time?
* update all items' read statuses to match tags
    * manual option
    * automatically on startup? Or is there some app->desktop sync operation I could use to trigger this?

**Comment by bruce-willis** [comment]
> it seems that advanced search isn't possible on the Zotero app at all

Yes, it seems like this feature is still not implemented in the iOS app. Saved searches also do not exist.

I've tested using the following collection (six items, one without any status and with all five statuses):  
<img width="1204" alt="image" src="https://github.com/user-attachments/assets/cedbb2a2-d5b0-458b-a318-d195512d48f6">

I've tried to use search bar to search for status in the `Extra` field, but maybe the titles are also playing role in the experiment.


1. `Read_Status: New` — (`New`, `To Read`, `In Progress`, `Not Reading` statuses are shown ❌) 
<details><summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/0f5b1a5d-d598-4409-a3dc-86e8cd6f600a">
</details> 

2. `Read_Status: To Read` — (`New`, `To Read`, `In Progress`, `Read` , `Not Reading` statuses are shown ❌)
<details><summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/f88067ee-ea3f-437f-aedb-000f94a580c4">
</details> 

3. `Read_Status: In Progress` — (`New`,  `In Progress` statuses are shown ❌)
<details><summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/e08f28e8-6b20-45ca-b0ad-c63b5e6495a8">
</details> 

4. `Read_Status: Read` — (`New`, `To Read`, `In Progress`, `Read` ,  `Not Reading` statuses are shown ❌)
<details><summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/3b7ace43-3229-4db3-8661-cd093bfc4438">
</details> 

5. `Read_Status: Not Reading` — (only `Not Reading` status is shown ✅)
<details><summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/dcb30ea3-1ccd-4b89-909f-34880500673a">
</details> 

***

> But filtering by tags is possible? Also the non-coloured tags? 

Yes, it works as expected. I manually created tags (just for these 5 items) and it is possible to filter.

<details><summary>screenshots</summary>
<img src="https://github.com/user-attachments/assets/3adb8721-7d42-425d-b964-e034446bc86b">
<img src="https://github.com/user-attachments/assets/f70b7cfb-c09e-4141-b924-242101fbeeba">
</details>

**Comment by bruce-willis** [comment]
> I've started work on this - any thoughts would be great!

Thanks! Let me know if you need a beta-tester :D 

I also think that this feature should be optional, since I suppose there are users who don't really need it (e.g. only using from the desktop app).

Also tags [support](https://forums.zotero.org/discussion/80012/how-to-add-emojis-in-the-tag-fields) not only colours, but also emoji:

<details><summary>iPad</summary>
<img src="https://github.com/user-attachments/assets/fff04a1f-c56b-409f-88bd-1722773d54a4">
</details> 

<details><summary>desktop</summary>
<img src="https://github.com/user-attachments/assets/acfc44a8-8eb4-4346-9a87-0a5b5d5aa40b">
</details> 

I have never developed anything for Zotero, but maybe [it is possible](https://arc.net/l/quote/ohofzeyj) to get when the tag was set?

**Comment by Dominic-DallOsto** [comment]
Ok, if you have time to checkout the v1.4.0-beta release that would be great!

There's an option to enable this feature which will prompt you to update all your items' tags to match their read statuses.

After that any time you set and item's read status, its tag will be updated to match.

At the moment if you change an item's tag on the app, you need to manually run the option to update read statuses from tags in the settings menu.
* can you explain a bit more about how the app <-> desktop Zotero synchronisation works? I wonder if there's a natural point to run the tag -> read status update automatically?

Also, could you test whether the modification date for an item gets updated based on when you change tags in the app?

1. Choose an item in the desktop app and selected it
2. Open the error console: Tools -> Developer -> Error Console
3. Type `ZoteroPane.getSelectedItems()[0]._dateModified ` in the console and press enter
4. Change the item's tag on the app, synchronise it to the desktop, then run the command again
5. Does the date modified change? If so, does it reflect when you changed the tag in the app, or when you synchronised the item to the desktop?

**Comment by Dominic-DallOsto** [comment]
I just updated the beta release to test this feature so it's based off the latest version of this plugin.

If you have the chance to test it and provide any feedback that'd be great!

**Comment by MaxVonMustermann** [comment]
Works like a charm!
The only addition for further polishing: Is it possible to show tag icons (emojis) in the right click context menu on the desktop Zotero app? Currently, right clicking an item, choosing Read Status and then opening the selection menu shows plain text only.

Thank you in advance!
